File: LeetCode/201-300/206.反转链表.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {
   ListNode p = head;
        ListNode prev = null;
        while (p != null) {
            ListNode cur = p.next;
            p.next = prev;
            prev = p;
            p = cur;
        }
        return prev;
    }
}

File: LeetCode/201-300/206. 反转链表.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
 func reverseList(head *ListNode) *ListNode {
	p := head
	var prev *ListNode
	for p != nil {
		next := p.Next
		p.Next = prev
		prev = p
		p = next
	}
	return prev
}